api/src/main/java/org/openmrs/Obs.java

```diff
@@ -106,55 +140,92 @@
 	 * 
 	 * @see #isObsGrouping() (??)
 	 */
+	@JoinColumn(name = "obs_group_id")
+	@ManyToOne
 	protected Obs obsGroup;
 	
 	/**
 	 * The list of obs grouped under this obs.
 	 */
 	@AllowDirectAccess
+	@OneToMany(mappedBy = "obsGroup")
+	@Cascade(CascadeType.DELETE)
+	@OrderBy("obs_id")
+	@BatchSize(size = 25)
+	@Access(AccessType.FIELD)
 	protected Set<Obs> groupMembers;
 	
+	@JoinColumn(name = "value_coded")
+	@ManyToOne
 	protected Concept valueCoded;
 	
+	@JoinColumn(name = "value_coded_name_id")
+	@ManyToOne
 	protected ConceptName valueCodedName;
 	
+	@JoinColumn(name = "value_drug")
+	@ManyToOne
 	protected Drug valueDrug;
 	
+	@Column(name = "value_group_id")
 	protected Integer valueGroupId;
 	
+	@Column(name = "value_datetime", length = 19)
 	protected Date valueDatetime;
 	
+	@Column(name = "value_numeric", length = 22)
 	protected Double valueNumeric;
 	
+	@Column(name = "value_modifier", length = 2)
 	protected String valueModifier;
 	
+	@Column(name = "value_text", length = 65535)
 	protected String valueText;
 	
+	@Column(name = "value_complex")
 	protected String valueComplex;
 	
 	// ComplexData is not persisted in the database.
 	protected transient ComplexData complexData;
 	
+	@Column(name = "comments")
 	protected String comment;
 	
-	protected transient Integer personId;
+	@Column(name = "person_id", nullable = false, insertable = false, updatable = false)
+	protected Integer personId;
 	
+	@JoinColumn(name = "person_id")
+	@ManyToOne(optional = false)
 	protected Person person;
 	
+	@JoinColumn(name = "order_id")
+	@ManyToOne
 	protected Order order;
 	
+	@JoinColumn(name = "location_id")
+	@ManyToOne
 	protected Location location;
 	
+	@JoinColumn(name = "encounter_id")
+	@ManyToOne
 	protected Encounter encounter;
 	
+	@JoinColumn(name = "previous_version", unique = true)
+	@ManyToOne
 	private Obs previousVersion;
 	
+	@Column(name = "form_namespace_and_path")
+	@Access(AccessType.FIELD)
```

Comment:
Could you remind me why we explicitly set the access type for this?

Reply:
I was just replicating the exact mapping in the hbm file, it was already accessed by field in xml file.

As to why it was historically mapped this way, if I recall well, it was because the class exposes namespace and formfield path as the public properties but both are stored in one column as concatenation separated by some character, therefore formNamespaceAndPath is internal logic.